Use logarithmic values for score

maudlin27 opened the pull request:
See the suggestion here:
https://forum.faforever.com/topic/6256/option-for-logarithmic-scaling-in-the-score-board?_=1690917013215

Having a logarithmic value for the score makes it easier to see how well players are performing throughout the game, since the score typically tends to increase closer to an exponentional value than a linear value as the game progresses.

This change uses logarithmic values to adjust how the graphs are displayed, and the y axis intervals, while still showing the original (pre-log) values.

maudlin27 commented:
One point to flag - I wasn't sure what the label in row 371 related to (it's not the mouse-over on the graph itself), but I've added code to convert the figure from a logarithmic value to be safe.

Garanas commented:
@maudlin27 do you know what base it uses, is it 10?

![image](https://github.com/FAForever/fa/assets/15778155/6517988b-9b2b-48ad-bc6e-1d1be1e10c48)

Garanas reviewed (changes requested):
I end up with quite a few graphs that look like this:

![image](https://github.com/FAForever/fa/assets/15778155/e625abd2-d75a-4e64-973f-65b5960280cb)

Which doesn't use the top part. Usually running a game for 15+ minutes turns a graph into this. I like the idea, but the scale is off here

maudlin27 commented:
Updated for the following:

- Moved logic adjusting the scaling into separate functions
- Switched from logarithmic scaling to using square root of the value - main reason is logarithmic would scale off too quickly, so the majority of the graph was taken up by the first minute of the score which is the least relevant part
- Adjusted y axis so most of the graph is used, and also made it 5% higher than the highest value so it's more likely the highest value labels appear within the graph
- Added commas for easier reading of large values

Illustrative example:
![image](https://github.com/FAForever/fa/assets/95254039/cf7bc7e5-1e2a-469d-be51-4d5675d4bb4c)

Garanas commented:
![image](https://github.com/FAForever/fa/assets/15778155/b1943455-3242-41fb-9e27-6da16511584f)

![image](https://github.com/FAForever/fa/assets/15778155/e88463e3-c7ba-4565-9303-e02646bc23ca)

![image](https://github.com/FAForever/fa/assets/15778155/9132c182-bc09-49b5-9c88-8bd1dee2a2ec)

Garanas commented on lua/ui/dialogs/hotstats.lua:
```diff
@@ -852,7 +886,7 @@ function create_graph(parent,path,x1,y1,x2,y2)
             end
             if posX()>x1 and posX()<x2 and posY()>y1 and posY()<y2 then
-                local  value = tps_format((posX()-x1)/(x2-x1)*scoreInterval*data_nbr) .. " / " .. math.floor(((y2-posY())/factor))
+                local  value = tps_format((posX()-x1)/(x2-x1)*scoreInterval*data_nbr) .. " / " .. FormatNumber(math.floor(ReverseScaling((y2-posY())/factor))) --This is the value that gets shown when we hover the mouse over any point in the graph, so want to do exponentional to reverse the logarithmic so we see the actual value
```
Can you update the comment here, it still mentions the exponentional / log approach

Garanas commented on lua/ui/dialogs/hotstats.lua:
```diff
@@ -476,6 +477,35 @@ function makeKMG(int)
+end
+
+function ReverseScaling(val)
+    --LOG('Val pre reverse='..val..'; expected val post reversal='..(val * val)..'; scaled value of this='..AdjustValueScale((val * val)))
```
Please remove commented code

maudlin27 commented:
Updated

Garanas commented:
Great work @maudlin27 !
